### PR TITLE
Allow run tests manually

### PR DESCRIPTION
**Pull request by dylant-da**
Tests are now run manually & individually in `daml studio`, so that you can use short-running tests along long-running ones.

https://github.com/digital-asset/daml/issues/15780

### CODE REVIEW

**Comment by stefanobaghino-da** [comment]
Probably a short video cast showcasing the feature could help the poor product people who can't read code. :trollface:

**Comment by dylant-da** [comment]
> Probably a short screencast showcasing the feature could help the poor product people who can't read code. :trollface:

I was working on this, OBS Studio won't behave :P

**Review by akrmn — APPROVED** [review]
Great work, thank you @dylant-da!

File: compiler/daml-extension/src/extension.ts [hunk 1]
```diff
@@ -406,6 +410,21 @@ namespace DamlVirtualResourceNoteNotification {
   );
 }
 
+interface VirtualResourceFormattingStartedParams {
+  /** The virtual resource uri */
+  uri: string;
+
+  /** A message describing the formatting */
+  message: string;
```
Comment (akrmn) [inline compiler/daml-extension/src/extension.ts:418]: this doesn't seem to be used

File: compiler/damlc/daml-rule-types/src/Development/IDE/Core/RuleTypes/Daml.hs [hunk 2]
```diff
@@ -204,12 +211,28 @@ instance Binary   RunScenarios
 instance Hashable RunScenarios
 instance NFData   RunScenarios
 
+data RunSingleScenario = RunSingleScenario T.Text
+    deriving (Eq, Show, Typeable, Generic)
+--data RunSingleScenario = RunSingleScenario
+--    deriving (Eq, Show, Typeable, Generic)
```
Comment (akrmn) [inline compiler/damlc/daml-rule-types/src/Development/IDE/Core/RuleTypes/Daml.hs:217]:
```suggestion
```

File: compiler/daml-extension/src/extension.ts (hunk 1 above)
Comment (dylant-da) [reply compiler/daml-extension/src/extension.ts:418]: You're right, this was meant for messaging relating to https://github.com/digital-asset/daml/issues/15782, I'll remove it from this PR for now

File: compiler/damlc/daml-rule-types/src/Development/IDE/Core/RuleTypes/Daml.hs (hunk 2 above)
Comment (dylant-da) [reply compiler/damlc/daml-rule-types/src/Development/IDE/Core/RuleTypes/Daml.hs]: fixed

File: compiler/damlc/daml-rule-types/src/Development/IDE/Core/RuleTypes/Daml.hs [hunk 3]
```diff
@@ -204,12 +211,28 @@ instance Binary   RunScenarios
...
 data RunScripts = RunScripts
     deriving (Eq, Show, Typeable, Generic)
 instance Binary   RunScripts
 instance Hashable RunScripts
 instance NFData   RunScripts
 
+data RunSingleScript = RunSingleScript T.Text
+    deriving (Eq, Show, Typeable, Generic)
+--data RunSingleScript = RunSingleScript
+--    deriving (Eq, Show, Typeable, Generic)
```
[the remark this answers is not in the document]
Comment (dylant-da) [reply compiler/damlc/daml-rule-types/src/Development/IDE/Core/RuleTypes/Daml.hs]: fixed

**Pull request merged**